api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -213,49 +215,46 @@ private CodeableConcept getManifestation(List<AllergyReaction> reactions) {
 	}
 	
 	private AllergyIntolerance.AllergyIntoleranceSeverity getFhirSeverity(Concept severityConcept) {
-		if (severityConcept == null) {
+		getSeverityConceptUuids();
+		
+		if (severityConcept == null || severityConceptUuids.size() < 4) {
 			return null;
```

Comment:
You should be checking if the severityConceptUuids array is not empty

Reply:
```java
globalPropertyService.getGlobalProperties(FhirConstants.GLOBAL_PROPERTY_MILD,
		    FhirConstants.GLOBAL_PROPERTY_MODERATE, FhirConstants.GLOBAL_PROPERTY_SEVERE,
		    FhirConstants.GLOBAL_PROPERTY_OTHER);
```
This method filters out null global properties. So there is a probability that size of results will not be 4 always